.github/workflows/check-3rd-party.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+FILE=$1
+status=0
+
+while read -r line; do
+  src=`echo $line | sed -E "s/(.*)uses://g" | xargs`
```

Comment:
Why the xargs?

Also, this is going to be pretty brittle. Can we guarantee, e.g., the `yq` command (or something that can read yaml)?

Reply:
Another option would be a language that can do the yaml parsing and string manipulation all at once, like a small python or go program. I think this is ultimately up to @marten-seemann's choice, depending on what software is readily available and does not require extra downloading/installing.

Reply:
I was trying to avoid baking in the assumption that all workflows will be valid yaml. If we decide to use a templating system later, we might end up with something that yaml parser won't like.

Reply:
Ok, fair enough. Maybe `sed -ne 's/[[:space:]]*uses:[[:space:]]*//p`? (can also skip the grep this way)

Reply:
Small modification: `sed -ne 's/[[:space:]-]*uses:[[:space:]]*//p'` also removes the (optional) `-`.